core/src/main/java/org/infinispan/conflict/EntryMergePolicyFactoryRegistry.java

```diff
@@ -0,0 +1,39 @@
+package org.infinispan.conflict;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.infinispan.configuration.cache.PartitionHandlingConfiguration;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+
+@Scope(Scopes.GLOBAL)
+public class EntryMergePolicyFactoryRegistry {
+
+   private final List<EntryMergePolicyFactory> factories = new ArrayList<>();
```

Comment:
Shouldn't we have just one "eager" factory and one "lazy" factory?

Seeing the list of factories here made me think that each deployment would have its own factory, and not knowing much about server deployments I had to see the wait of up to 60 seconds in `DeployedMergePolicyFactory#createInstance()` to realize I was wrong :)

Or maybe add a comment saying that we expect a single factory in embedded mode and an additional factory in the server.

Reply:
I have added to comments to the registry to explain the intended usage.